### PR TITLE
fix: Delete videos on disk directly, mark for deletion on next refresh if that fails

### PR DESCRIPTION
**Pull request by StephixOne**
Attempts a fix for [Issue#506](https://github.com/aza547/wow-recorder/issues/506).

My Windows does not complain when deleting a file that may be open, even if I've played half of it in warcraftrecorder, it'll delete it just fine. As such, simply deleting the file seems to do the trick. I've also added a fallback to the old mechanism, of marking the files for deletion and letting the next refresh/restart take care of it, if the hard delete fails.

I've also reset the video player progress when a deletion happens, because being `x` seconds through a video, deleting it, then landing at `x` seconds in the one it drops you to felt a bit jarring. Probably not the most common use-case, but it shouldn't hurt.

### CODE REVIEW

File: src/main/Manager.ts [hunk 1]
```diff
@@ -127,7 +128,8 @@ export default class Manager {
       valid: false,
       current: this.obsBaseCfg,
       get: (cfg: ConfigService) => getObsBaseConfig(cfg),
-      validate: async (config: ObsBaseConfig) => Manager.validateBaseCfg(config),
+      validate: async (config: ObsBaseConfig) =>
```
Comment (aza547) [inline src/main/Manager.ts:131]: I think your prettier is somehow ignoring the `/* eslint-disable prettier/prettier */` above here?

**Comment by aza547** [comment]
Looks great - more than happy with the compromise / fallback behaviour - no need to let perfect be the enemy of good.

Happy to merge / approve once the prettier / yarn stuff is fixed.

File: src/main/Manager.ts (hunk 1 above)
Comment (StephixOne) [reply src/main/Manager.ts:131]: Let me check on that.

File: package-lock.json
Comment (aza547) [inline package-lock.json:1]: Still a huge diff here that I think you intended to revert?

Comment (StephixOne) [reply package-lock.json:1]: My `npm` does that, as yarn uses its own lockfile, `yarn.lock`. What node version are you using?

Comment (aza547) [reply package-lock.json:1]: I have a global npmrc messing with this. 

https://stackoverflow.com/questions/55384977/npm-install-removes-dev-true-from-several-packages-in-package-lock-json

Agree it's fine.

**Comment by aza547** [comment]
Thanks for the fix!

**Pull request merged**